enhancements/sig-architecture/29-manifestwork-status-feedback/README.md

```diff
@@ -0,0 +1,243 @@
+
+```yaml
+
+kind: ManifestWork
+metadata:
+  name: demo-work1
+spec:
+  workload:
+    manifests:
+    - apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        name: hello
+        namespace: default
+      spec:
+        selector:
+          matchLabels:
+            app: hello
+        template:
+          metadata:
+            labels:
+              app: hello
+          spec:
+            containers:
+            - name: hello
+              image: quay.io/asmacdo/busybox
+              command: ['sh', '-c', 'echo "Hello, World!" && sleep 3600']
+  statusFeedbackOptions:
+  - resourceIdentifier:
+      group: apps
+      resource: deployments
+      name: hello
+      namespace: default
+    type: IntrerestedFields
+    intrestedFields:
+    - name: availableReplicas
+      jsonPath: .availableReplicas
+    - name: availableCondition
+      jsonPath: .conditions[?(@.type=="Available")].status
+```
+
+The status of `ManifestWork` will be updated to add
+
+```go
+
+// ManifestCondition represents the conditions of the resources deployed on a
+// managed cluster.
+type ManifestCondition struct {
+	// ResourceMeta represents the group, version, kind, name and namespace of a resoure.
+	// +required
+	ResourceMeta ManifestResourceMeta `json:"resourceMeta"`
+
+	// StatusFeedback represents the values of the status feedback
+	StatusFeedback StatusFeedback `json:"statusFeeback,omitempty"`
+
+	// Conditions represents the conditions of this resource on a managed cluster.
+	// +required
+	Conditions []metav1.Condition `json:"conditions"`
+}
+
+type StatusFeedback struct {
+    // InterestedValue represents the value of the interested field
+    InterestedValue []InterestedValue string `json:"intrestedValue"`
+}
+
+type InterestedValue struct {
+    // name represents the aliase name for this field
+	// +required
+	Name string `json:"name"`
+
+    // value represents the value of the field
+	Value FieldValue `json:"fieldValue"`
+}
+
+type FieldValue struct {
+    // Type is the type of the value, it can by integer, string or boolean
+	Type ValueType `json:"type"`
+
+	// +optional
+	Integer int32 `json:"integer,omitempty"`
+
+	// +optional
+	String string `json:"string,omitempty"`
+
+	// +optional
+	Boolean bool `json:"boolean,omitempty"`
+}
+
+```
+
+So the work agent will return a status with the above example as
+
+```yaml
+status:
+  resourceStatus:
+    manifests:
+      - conditions:
+        - lastTransitionTime: "2021-10-14T14:59:09Z"
+          message: Apply manifest complete
+          reason: AppliedManifestComplete
+          status: "True"
+          type: Applied
+        - lastTransitionTime: "2021-10-14T14:59:09Z"
+          message: Resource is available
+          reason: ResourceAvailable
+          status: "True"
+          type: Available
+        statusFeeback:
+          interestedValues:
+          - fieldValue:
+              integer: 1
+              type: integer
```

Comment:
IIUC all the feedbacks will be treated as plain yaml content for any kind of resource. currently supported types in yaml are:

- null
- boolean
- float
- integer
- string
- list
- map

i think we dont need to deal w/ int-or-string specifically. just clarify the failure path when the expected type in the spec doesnt match the actual feedback will be sufficient. e.g. say if we asserted `boolean` but the result turns out to be a string, then how should that be reflected in the status? my first thought is that we should explicitly show the failures to the feedback consumers, the feedback should deliver the message accurately otherwise it can be extra complex for a consumer (e.g. a multi-cluster operator) to read the feedback.

Reply:
Added a section to indicate error checking.